Examples/Io/Root/src/RootPropagationStepsWriter.cpp

```diff
@@ -127,37 +127,21 @@ ActsExamples::ProcessCode ActsExamples::RootPropagationStepsWriter::writeT(
     m_step_usr.clear();
     m_nStepTrials.clear();
 
-    // loop over single steps
+    // Loop over single steps
     for (auto& step : steps) {
-      // the identification of the step
-      Acts::GeometryIdentifier::Value volumeID = 0;
-      Acts::GeometryIdentifier::Value boundaryID = 0;
-      Acts::GeometryIdentifier::Value layerID = 0;
-      Acts::GeometryIdentifier::Value approachID = 0;
-      Acts::GeometryIdentifier::Value sensitiveID = 0;
+      const auto& geoID = step.geoID;
+      m_sensitiveID.push_back(geoID.sensitive());
+      m_approachID.push_back(geoID.approach());
+      m_layerID.push_back(geoID.layer());
+      m_boundaryID.push_back(geoID.boundary());
+      m_volumeID.push_back(geoID.volume());
+
       int material = 0;
-      // get the identification from the surface first
       if (step.surface) {
-        auto geoID = step.surface->geometryId();
-        volumeID = geoID.volume();
-        boundaryID = geoID.boundary();
-        layerID = geoID.layer();
-        approachID = geoID.approach();
-        sensitiveID = geoID.sensitive();
         if (step.surface->surfaceMaterial() != nullptr) {
           material = 1;
         }
       }
-      // a current volume overwrites the surface tagged one
-      if (step.volume != nullptr) {
-        volumeID = step.volume->geometryId().volume();
-      }
```

Comment:
Hey @paulgessinger  - this is the changed logic, the step writer actually chose to overwrite the volume identification value from the surface, which is ok as well, it just didn't puts logic into a writer ... and if another writer does this differently it will never be caught.